src/main/java/com/streamr/client/utils/OrderedMsgChain.java

```diff
@@ -69,6 +84,12 @@ public synchronized void add(StreamMessage unorderedMsg) {
             // Prevent memory exhaustion under unusual conditions by limiting the queue size
             if (queue.size() < MAX_QUEUE_SIZE) {
                 queue.offer(unorderedMsg);
+            } else if (skipGapsOnFullQueue) {
+                log.warn("Queue is full. Emptying and processing new mesage.");
+                gap.cancel();
```

Comment:
Should this and the below line be replaced with `clearGap()`? Should we add `gap.purge()` to `clearGap()`?